src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -817,7 +891,9 @@ protected BoundExpression BindInferredVariableInitializer(DiagnosticBag diagnost
             TypeSymbol declTypeOpt,
             AliasSymbol aliasOpt,
             DiagnosticBag diagnostics,
-            CSharpSyntaxNode associatedSyntaxNode = null)
+            CSharpSyntaxNode associatedSyntaxNode = null,
+            MethodSymbol disposeMethod = default, 
```

Comment:
You shouldn't need the dispose or conversion after you remove them from `BindVariableDeclaration`